src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java

```diff
@@ -120,18 +120,19 @@
  * property on the java launcher command line. Setting it at runtime with
```

Comment:
I may have missed it but does LocalServiceProvider's description of "java.locale.providers" say what the value of the system property is? If providers don't have names then I'm wondering if this system property can be used to select control ordering if I deploy my own provider. If not, then it makes me wonder if the definition of this system property needs to move to the implNote section.

Reply:
Locale providers provided by users can all be loaded in the name of `SPI`, as they are the real implementation of `LocaleServiceProvider` class, so the order of the preference can be specified against JDK's `CLDR` provider. Does this answer your question?

Reply:
> Locale providers provided by users can all be loaded in the name of `SPI`, as they are the real implementation of `LocaleServiceProvider` class, so the order of the preference can be specified against JDK's `CLDR` provider. Does this answer your question?

There are two parts to this PR. One part is the deprecation of "COMPAT" with a warning if used, no issue there. The second part changes the text from "The JDK Reference Implementation .." to the end of the class description to be an implNote.  My concern is that is the system property "java.locale.providers" is introduced in the normative section but it doesn't say anything about values. It just says that it can be used to configured the user's preferred order. The possible names and everything about ordering is in the implNote section. To make this work is going to need some expansion of the text in the normative section to say that the value is a sequence of strings, separated by a comma. It will also need to say something about the names as they are implementation specific. Also what happens if I deploy two custom locale providers on the class path, which one is used as I don't think the system property covers that.

Maybe the right thing here is to drop "@implNote" (just the tag) from this PR so that the deprecation, warning, and the small adjustments to the class description are pushed out of the way. A follow-up PR could re-examine the class description and the description of the "java.locale.providers" property.

Reply:
> Good point. Removed the @implNote tag for now and filed a separate issue to clarify the system property: https://bugs.openjdk.org/browse/JDK-8305595

Okay, that works for me.